processing/src/main/java/io/druid/query/filter/InDimFilter.java

```diff
@@ -117,6 +121,46 @@ public ExtractionFn getExtractionFn()
   @Override
   public DimFilter optimize()
   {
+    InDimFilter inFilter = optimizeLookup();
+    if (inFilter.values.size() == 1) {
```

Comment:
there can be an NPE here, but I thikn we shold just add a preconditions check in the constructor for null values